@navikt/core/react/src/layout/base/BasePrimitive.tsx

```diff
@@ -0,0 +1,197 @@
+import cl from "clsx";
+import React from "react";
+import { Slot } from "../../util/Slot";
+import { getResponsiveProps, getResponsiveValue } from "../utilities/css";
+import { ResponsiveProp, SpacingScale } from "../utilities/types";
+
+export interface BasePrimitiveProps {
+  children: React.ReactElement;
+  className?: string;
+  /** Padding around children. Accepts a spacing token or an object of spacing tokens for different breakpoints.
+   * @example
+   * padding='4'
+   * padding={{xs: '2', sm: '3', md: '4', lg: '5', xl: '6'}}
+   */
+  padding?: ResponsiveProp<SpacingScale>;
+  /** Horizontal padding around children. Accepts a spacing token or an object of spacing tokens for different breakpoints.
+   * @example
+   * paddingInline='4'
+   * paddingInline='4 5'
+   * paddingInline={{xs: '0 32', sm: '3', md: '4 5', lg: '5', xl: '6'}}
+   */
+  paddingInline?: ResponsiveProp<
+    SpacingScale | `${SpacingScale} ${SpacingScale}`
+  >;
+  /** Vertical padding around children. Accepts a spacing token or an object of spacing tokens for different breakpoints.
+   * @example
+   * paddingBlock='4'
+   * paddingBlock='4 5'
+   * paddingBlock={{xs: '2', sm: '3', md: '4', lg: '5', xl: '6'}}
+   */
+  paddingBlock?: ResponsiveProp<
+    SpacingScale | `${SpacingScale} ${SpacingScale}`
+  >;
+  /**
+   * CSS `width`
+   */
+  width?: ResponsiveProp<string>;
+  /**
+   * CSS `min-width`
+   */
+  minWidth?: ResponsiveProp<string>;
+  /**
+   * CSS `max-width`
+   */
+  maxWidth?: ResponsiveProp<string>;
+  /**
+   * CSS `height`
+   */
+  height?: ResponsiveProp<string>;
+  /**
+   * CSS `min-height`
+   */
+  minHeight?: ResponsiveProp<string>;
+  /**
+   * CSS `max-height`
+   */
+  maxHeight?: ResponsiveProp<string>;
+  /**
+   * CSS `position`
+   */
+  position?: ResponsiveProp<
+    "static" | "relative" | "absolute" | "fixed" | "sticky"
+  >;
+  /**
+   * CSS `inset`
+   */
+  inset?: ResponsiveProp<SpacingScale>;
+  /**
+   * CSS `top`
+   */
+  top?: ResponsiveProp<SpacingScale>;
+  /**
+   * CSS `right`
+   */
+  right?: ResponsiveProp<SpacingScale>;
+  /**
+   * CSS `bottom`
+   */
+  bottom?: ResponsiveProp<SpacingScale>;
+  /**
+   * CSS `left`
+   */
+  left?: ResponsiveProp<SpacingScale>;
+  /**
+   * CSS `overflow`
+   */
+  overflow?: ResponsiveProp<"visible" | "hidden" | "clip" | "scroll" | "auto">;
```

Comment:
We should revisit the JSDOC later. It's a little inconsistent that we write about the object option only on some of the props, when it applies to all of them. Maybe the JSDOC for each prop should be really simple and short, with a link to some place with more details 🤔

Reply:
Sounds like a good idea, think `ResponsiveProp` and in general the Primitives concept deserves a documentation-page under /grunnleggende when its implemented and tested. Can link to that documentation then 👍